src/lib/EventListener/CredentialsExpirationWarningListener.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\EventListener;
+
+use DateTime;
+use DateTimeInterface;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\Core\MVC\Symfony\Security\UserInterface;
+use EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface;
+use EzSystems\EzPlatformAdminUi\Specification\SiteAccess\IsAdmin;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Http\SecurityEvents;
+use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class CredentialsExpirationWarningListener implements EventSubscriberInterface
+{
+    /** @var \EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface */
+    private $notificationHandler;
+
+    /** @var \Symfony\Component\Translation\TranslatorInterface */
+    private $translator;
+
+    /** @var \Symfony\Component\Routing\Generator\UrlGeneratorInterface */
+    private $urlGenerator;
+
+    /** @var \eZ\Publish\API\Repository\UserService */
+    private $userService;
+
+    /** @var array */
```

Comment:
```suggestion
    /** @var string[] */
```

Reply:
Actually, the correct type hint is `string[][]` 😉 

![image](https://user-images.githubusercontent.com/211967/63835645-2d9c7e80-c978-11e9-9094-d1de0931beba.png)

Reply:
Fixed in https://github.com/ezsystems/ezplatform-admin-ui/pull/1060/commits/6ef932bb76c280e35cd453553fcfabbf15dae82b